src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNICommon.cs

```diff
@@ -59,10 +62,17 @@ public void Read(byte[] bytes)
         {
             SMID = bytes[0];
             flags = bytes[1];
+#if NETCOREAPP
+            sessionId = BinaryPrimitives.ReadUInt16LittleEndian(new ReadOnlySpan<byte>(bytes, 2, 2));
```

Comment:
nit: `bytes.AsSpan(2, 2)` would be a bit more readable in my opinion, these lines are already too long.